### DISCUSSION TITLE
SB8: storybook doesn't open previous browser tab, open new one every time

### DISCUSSION THREAD

**Discussion by Freytag** [storybookjs/storybook#28791] [Help]
### Summary

just did and upgrade to SB8 
In SB7 when i run storybook after stopping it, If i still had a tab open with `localhost:6006` from a previous session it would just focus that tab. 
Now in SB8 when I run storybook it opens a new browser tab instead of going to the one that is open.

**Comment by Marklb** [comment]
Docs: https://storybook.js.org/docs/api/cli-options#api-commands

If using CLI:
```sh
storybook dev --no-open
```

If using Angular, use cli or add `"open": false` to the builder in `angular.json`.

**Reply by Freytag** [reply]
This prevents it from opening the a browser any time. I want it to not open _new_ windows each time and instead just focus the window that is already open.

**Reply by Marklb** [reply]
What OS and browser are you using?

I didn't know it ever did that, but I just checked and it does seem like that is a feature supported by the two libraries used. From what I see, that may be something that only works on macOS and I don't own a Mac, which would explain why I never saw it do that.

The only thing I see that you could try is setting `OPEN_MATCH_HOST_ONLY=true` environment variable. ex: `OPEN_MATCH_HOST_ONLY=true storybook dev`

Maybe someone more familiar with how opening the browser works will answer, but if you want to look at how it works this should be the file: https://github.com/storybookjs/storybook/blob/next/code/core/src/core-server/utils/open-in-browser.ts

**Reply by migueln1987** [reply]
The problem is in the file that you provided, first of all why make the assumption that the environment variable for BROWSER is set for anyone. I don't have it set I'm also on macOS. Second of all, why the check for chromium based browsers individually? There are more chromium browsers based browsers than the ones listed. The documentation for the open function tells one exactly what it does

```
Open a new instance of the app even it's already running.

A new instance is always opened on other platforms.
```
The solution should not be that "we need to set more environment variables" when the one assumption you made is that we have an environment variable made. This is already a solved problem, every other server can refresh the open tab

**Reply by Marklb** [reply]
I don't think there is an assumption that `BROWSER` is set. `@aw-web-design/x-default-browser` is also used, which does different checks.

As for why check for chromium based browsers individually, I don't know if there is a generic way to determine chromium-based. `better-opn` has a check with more browsers, so I don't know why Storybook needs to also do a check, though.

`open` should open a new instance, as you said. My assumption was that if it used to reuse the same tab, and that code hasn't changed in a while, then the conditions that cause `better-opn` to get called are most likely met. Since I don't have a Mac to test on, I was just suggesting [`OPEN_MATCH_HOST_ONLY`](https://www.npmjs.com/package/better-opn#reuse-tab-by-match-host) mainly as a potential workaround solution. If the environment variable worked, that may point out where to look for someone that wants to submit a PR fixing it.

**Comment by kano-lvgs** [comment]
I'm waiting for fixing this problem, too ! I think it would be improved in future, because it was improved previous version.
https://github.com/storybookjs/storybook/issues/10328
